tests/basilisp/test_io.lpy

```diff
@@ -118,6 +119,10 @@
                                        *http-port*
                                        "/reader-http-req.txt"))))
         (finally
+          ;; A small wait is intentially added to accomodate for a
+          ;; potential delay in releasing the file by the HTTP server
+          ;; on MS-Windows.
+          (time/sleep 1)
```

Comment:
Do we anticipate this could still fail a noticeable percent of the time?

Reply:
I had another look at this, and I think we don't really need to sleep for any amount of time, but rather give a chance to the http-server thread to run so that it releases the file handle, thus I converted this to `(time/sleep 0)`. This is my speculation by looking at the code. I don't anticipate this to fail intermittently .

Reply:
Sorry I made a mistake rerunning the job with `(time/sleep 0)`, it does fail with 0, and reverted it back to `(time/sleep 1)`.

I had a quick look at the [SimpleHTTPRequestHandler](https://github.com/python/cpython/blob/a443c310ac87f214164cb1e3f8af3f799668c867/Lib/http/server.py#L646) code, and I'm not sure why this delay is needed, the file appears to be closed on serve (though I haven't stepped into this code to confirm this).

Not sure if you're concern is about the sleep interval of 1 sec is low? i can then try to attempt to delete every second for say 5 seconds in total.

Reply:
I just don't want to have a flaky test if it can be avoided. We can try it with this sleep and see how things go.

Reply:
Sure, I should investigate deeper if this turns out to be flaky.

Reply:
Ok thanks. We'll just try it out for now.